lib/services/cognitiveServicesSearch/lib/entitySearch/entitySearchAPIClient.d.ts

```diff
@@ -30,13 +31,25 @@ declare class EntitySearchAPI extends ServiceClient {
    *
    * @param {boolean} [options.noRetryPolicy] - If set to true, turn off default retry policy
    *
+   * @param {string} [options.acceptLanguage] - Gets or sets the preferred language for the response.
+   *
+   * @param {number} [options.longRunningOperationRetryTimeout] - Gets or sets the retry timeout in seconds for Long Running Operations. Default value is 30.
+   *
+   * @param {boolean} [options.generateClientRequestId] - When set to true a unique x-ms-client-request-id value is generated and included in each request. Default is true.
+   *
    */
-  constructor(credentials: ServiceClientCredentials, baseUri?: string, options?: ServiceClientOptions);
+  constructor(credentials: ServiceClientCredentials, baseUri?: string, options?: AzureServiceClientOptions);
```

Comment:
This line tells us that Tony Jin generated the EntitySearch client using the vanilla flavor of the node.js generator `--azure-arm` not provided as a parameter while generating via autorest.
Can you please confirm this with @lmazuel, whether he is using the vanilla flavor or the azure flavor for generating cognitive services credentials.

Even when you generate with the vanilla flavor we still need to add `--add-credentials=true` switch while generating via autorest because the client takes the credentials object as the first parameter in the constructor.

So I am not really sure if generating with the azure flavor would make a huge difference.

@lmazuel  - what do you think?

Reply:
@amarzavery vanilla I used!
Probably ARM would work, but I feel that philosophically causes me trouble (it's not ARM :p)

https://github.com/Azure/azure-sdk-for-python/blob/dbac87cd22bf888f9bd55eee0bf783e8e1e824af/swagger_to_sdk_config.json#L90-L92